src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -98,6 +127,58 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         _defaultAppearanceViewModel.IsDefault(true);
     }
 
+    void ProfileViewModel::_UpdateBuiltInIcons()
+    {
+        _BuiltInIcons = single_threaded_vector<IInspectable>();

```

Comment:
fwiw right now we do this once per profile when we could do it once per settings UI

Reply:
Nah, `_BuiltInIcons` is static so it's a shared instance across all ProfileVMs.

And we only call it twice:
https://github.com/microsoft/terminal/blob/bb328969730ed5a04db24f07e504fe1584d962e0/src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp#L116-L119

https://github.com/microsoft/terminal/blob/bb328969730ed5a04db24f07e504fe1584d962e0/src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp#L165-L168

Both of which are wrapped in a null-check.

Confirmed w/ breakpoint in `_UpdateBuiltInIcons()` and poking around SUI (and closing/reopening SUI).

Reply:
Oh jeez you're a step ahead of me!